crunchy_test.go

```diff
@@ -13,16 +13,25 @@ import (
 	"crypto/sha256"
 	"crypto/sha512"
 	"hash"
-	"strconv"
 	"testing"
 )
 
 var (
+	// example of valid and invalid passwords
+	pass = struct {
+		valid   string
+		invalid string
+	}{"d1924ce3d0510b2b2b4604c99453e2e1", ""}
+
 	pws = []struct {
 		pw       string
 		expected error
 		rating   uint
 	}{
+		// include values from pass in tests
```

Comment:
What's the purpose of including these values here (again)?

Reply:
Having a reference to a password known to be valid is useful for benchmarks because they never exit earlier and need to go though every check we perform.

We could search the `pws` slice for a valid password when we need one, or make a convention that the first password of the slice is valid. I thought being explicit would make things more clear. I.e. `pass.valid vs pws[0].pw`

The values of the `pass` struct are then inserted into the `pws` slice to make sure they remain valid. If a future change caused `pass.valid` to be invalid, tests would fail, indicating that benchmark numbers are not trustworthy.